docs/content/guides/dagster-pipes/subprocess/reference.mdx

```diff
@@ -0,0 +1,242 @@
+---
+title: "Modifying existing code to work with Dagster Pipes | Dagster Docs"
+description: "This page shows ways to execute external code with Dagster Pipes with different entities in the Dagster systen."
+---
+
+# Reference
+
+This reference shows usage of Dagster Pipes with other entities in the Dagster system. For a step-by-step walkthrough, refer to the [Dagster Pipes tutorial](/guides/dagster-pipes/subprocess).
+
+## Specifying environment variables and extras
+
+When launching the subprocess, you may want to make environment variables or additoional parameters available in the external process. Extras can be arbitrary, user-defined parameters made available on the context object in the external process.
+
+<TabGroup>
+<TabItem name="Provide in the Dagster code">
+
+The `run` method to the `PipesSubprocessClient` resource also accepts `env` and `extras` , which allow you to specify environment variables and extra arguments when executing the subprocess:
```

Comment:
Can we use PyObject here for `PipesSubprocessClient`?

Reply:
https://dagsterlabs.slack.com/archives/C05M31J51J6/p1696976745913529
`PipesSubprocessClient`  has not been added to the apidoc. will update once that's fixed by sean